Codec/Archive/Zip.hs

```diff
@@ -173,6 +174,10 @@ import Data.Word (Word16, Word32)
 import System.Directory
 import System.FilePath ((</>))
 import qualified System.FilePath as FP
+#ifndef _WIN32
```

Comment:
Where does `_WIN32` come from?

Reply:
I have seen something like `#if defined(mingw32_HOST_OS)` used to check if it is windows. How `_WIN32` is different and why it is preferable?

Reply:
I'll switch to `mingw32_HOST_OS`, thanks! `_WIN32` comes from gcc, and it's less reliable here.